### PR TITLE
Max/site subscribe

### LINKED ISSUE #421: Add Site Subscription
How are we actually doing this?

### PR DESCRIPTION
**Pull request by maxwolffe**
:skull:  Don't merge yet. 

Site subscription works. 
Waiting on Charles' PR to get merged to make UI changes and implement site un-subscription. 

Fixes #421

### CODE REVIEW

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Users/UserFragment.java [hunk 1]
```diff
@@ -88,9 +88,10 @@ private void initializeListLayout() {
         ArrayList<String> options = new ArrayList <String>();
 
         options.add("Field-Reports " + mUser.getFieldReportsCount());
+        options.add("Tasks " + mUser.getFieldReportsCount());
+        options.add("Sites " + mUser.getSitesCount());
         if (mUser.isEmployee() || mUser.isManager()){
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Users/UserFragment.java]: are these models implemented on the backend yet?

**Comment by maxwolffe** [comment]
Waiting on #444

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Users/UserFragment.java (hunk 1 above)
Comment (maxwolffe) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Users/UserFragment.java]: They aren't (except for Field report count I think), do we want this information to even be available though? Sites and Tasks we can access already. I may just take them out.

**Comment by maxwolffe** [comment]
Swag. Should work. :)

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/BaseRequest.java [hunk 2]
```diff
@@ -106,6 +106,8 @@ public static NetworkManager getNetworkManager(Context context) {
     @Override
     public HashMap<String, String> getHeaders() {
         HashMap<String, String> headers = new HashMap<String, String>();
+        Log.e("X-AUTH-TOKEN", preferences.getString("authentication_token", "none"));
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/BaseRequest.java]: Should remove these if they're just for debugging.

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteFragment.java [hunk 3]
```diff
@@ -246,4 +302,28 @@ public void onPause() {
...
+    private void markSubscribed(ArrayList<Site> sites) {
+        for (Site site : sites) {
+            if (site.getId().equals(mSite.getId())) {
+                FloatingActionButton subscribeButton = (FloatingActionButton) mView.findViewById(R.id.site_subscribe_site);
+                subscribeButton.setTitle("Unsubscribe from Site");
+                subscribeButton.setIcon(R.drawable.ic_bookmark_white_36dp);
+                mSubscribed = true;
+            }
+        }
+    }
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteFragment.java]: Wouldn't it be better to just check if the user is subscribed to this particular site?

Comment (maxwolffe) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteFragment.java]: yep. thanks for catching that, forgot to delete it after charles added the backend stuff.

**Pull request merged**